packages/@lwc/synthetic-shadow/src/polyfills/event-target/polyfill.ts

```diff
@@ -0,0 +1,155 @@
+    removeCustomElementEventListener,
+} from '../../faux-shadow/events';
+import {
+    isHostElement,
+    getHost,
+    SyntheticShadowRootInterface,
+} from '../../faux-shadow/shadow-root';
+import { eventTargetGetter } from '../../env/dom';
+import { isNodeShadowed } from '../../shared/node-ownership';
+import { contains } from '../../env/node';
+
+// this method returns true if an event can be seen by a particular eventTarget,
+// otherwise it returns false, which means the listener will never be invoked.
+function isQualifyingEventTarget(currentTarget: EventTarget, evt: Event): boolean {
+    const originalTarget = eventTargetGetter.call(evt);
+    if (originalTarget === currentTarget) {
+        // short-circuit when dispatched on the target directly
+        return true;
+    }
+    const { composed, bubbles } = evt;
+    if (!composed) {
+        if (bubbles) {
+            // bubbles true, composed false: only propagate up to the originalTarget's shadowRoot
+            if (originalTarget instanceof Node) {
+                if (isNodeShadowed(originalTarget)) {
+                    // if the original target is shadowed, every node in the path to its shadowRoot
+                    // will have visibility into the event, including slots since the target might
+                    // be slotted.
+                    const sr = originalTarget.getRootNode() as SyntheticShadowRootInterface;
+                    return contains.call(getHost(sr), currentTarget as Node);
+                }
+                // else it is a global node or document
+            }
+            // else it is window
+        } else {
+            // bubbles false, composed false: only dispatch it on the original target
+            return currentTarget === originalTarget;
+        }
+    } else if (!bubbles) {
+        // bubble false, composed true: only propagate via host elements
+        return isHostElement(currentTarget as Node);
+    }
+    // bubbles true, composed true
+    return true;
+}
+
+const EventListenerToWrapperMap: WeakMap<
+    EventListenerOrEventListenerObject,
+    EventListener
+> = new WeakMap();
+
+function getEventListenerWrapper(fnOrObj: EventListenerOrEventListenerObject): EventListener {
+    let wrapperFn: EventListener | undefined = EventListenerToWrapperMap.get(fnOrObj);
+    if (isUndefined(wrapperFn)) {
+        wrapperFn = function(this: EventTarget, e: Event) {
+            if (isQualifyingEventTarget(this, e)) {
+                if (typeof fnOrObj === 'function') {
+                    fnOrObj.call(this, e);
+                } else {
+                    fnOrObj.handleEvent && fnOrObj.handleEvent(e);
+                }
+            }
+        };
+        EventListenerToWrapperMap.set(fnOrObj, wrapperFn);
+    }
+    return wrapperFn;
+}
+
+// These methods are usually from EventTarget.prototype, but that's not available in IE11, the next best thing
+// is Node.prototype, which is an EventTarget as well.
+const {
```

Comment:
I believe this can be extracted in `../../env/node` and reuse it here. Node.prototype.addEventListener is not patched before this point.